content/en/blog/_posts/2021-04-08-cronjob-reaches-ga/index.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: blog
+title: 'CronJob Reaches GA'
+date: 2021-04-09
+slug: cronjob-reaches-ga
+---
+
+ **Authors:** Alay Patel (Red Hat), and Maciej Szulik (Red Hat)
+
+Since last summer, we have been working on a new CronJob Controller (Controller
+V2) implementation which initially landed in 1.20 as an alpha feature. In 
+kubernetes 1.21 the new controller is turned on by default, and the CronJob 
+resource is promoted to general availability (GA) (group version: `batch/v1`). This blog
+post walks through the driving forces behind this new development, gives a 
+brief description of a controller design in core kubernetes, and explains what
+users will gain with this new controller.
+
+The driving force behind promoting the API was this [proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/1635-prevent-permabeta#sig-apps). Its acceptance
+prevents APIs from being stuck in a permanent beta state. Over the years the
+old CronJob controller had received healthy feedback from the community and
+it had some well-known [issues](https://github.com/kubernetes/kubernetes/issues/82659).
+If this API was to be supported as GA, the controller would need a severe re-work.
+Instead, the SIG Apps community decided to introduce a new controller to
+gradually replace the old one. 
```

Comment:
```suggestion
The driving force behind promoting the API was Kubernetes' policy choice to
[ensure APIs move beyond beta](/blog/2020/08/21/moving-forward-from-beta/).
That policy aims to prevent APIs from being stuck in a “permanent beta” state.
Over the years the old CronJob controller implementation had received healthy feedback from
the community, with reports of several widely recognized
[issues](https://github.com/kubernetes/kubernetes/issues/82659).

If the beta API for CronJob was to be supported as GA, the existing controller code
would need substantial rework.
Instead, the SIG Apps community decided to introduce a new controller to
gradually replace the old one.
```